examples/restore_login.py

```diff
@@ -1,104 +1,119 @@
 #!/usr/bin/env python3
 
 import asyncio
-import getpass
 import json
 import os
-import sys
 
-import aiofiles
+from nio import (
+    AsyncClient,
+    AsyncClientConfig,
+    InviteMemberEvent,
+    LoginResponse,
+    MatrixInvitedRoom,
+    MatrixRoom,
+    RoomMessageText,
+    WhoamiResponse,
+)
 
-from nio import AsyncClient, LoginResponse
+DIR = os.path.dirname(__file__)
 
-CONFIG_FILE = "credentials.json"
+TOKEN_FILE = os.path.join(DIR, "credentials.json")
 
-# Check out main() below to see how it's done.
 
-
-def write_details_to_disk(resp: LoginResponse, homeserver) -> None:
-    """Writes the required login details to disk so we can log in later without
-    using a password.
-
-    Arguments:
-        resp {LoginResponse} -- the successful client login response.
-        homeserver -- URL of homeserver, e.g. "https://matrix.example.org"
+async def login(client: AsyncClient, password, device_name):
+    """
+    if possible, restore login token.
+    else login as new device.
     """
-    # open the config file in write-mode
-    with open(CONFIG_FILE, "w") as f:
-        # write the login details to disk
-        json.dump(
-            {
-                "homeserver": homeserver,  # e.g. "https://matrix.example.org"
-                "user_id": resp.user_id,  # e.g. "@user:example.org"
-                "device_id": resp.device_id,  # device ID, 10 uppercase letters
-                "access_token": resp.access_token,  # cryptogr. access token
-            },
-            f,
+    if not os.path.isfile(TOKEN_FILE):
```

Comment:
Should avoid blocking i/o (even something this minor in an example) in an async function; can use `aiofiles` (which is already a dependency) for this purpose.

Additionally, I prefer the use of `pathlib.Path` objects for path/file declaration/handling

Reply:
sure, i can improve that as well.